x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/googleaistudio/GoogleAiStudioStreamingProcessorTests.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.googleaistudio;
+
+import org.elasticsearch.common.xcontent.ChunkedToXContent;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.inference.external.response.googleaistudio.GoogleAiStudioCompletionResponseEntity;
+import org.elasticsearch.xpack.inference.external.response.streaming.ServerSentEvent;
+
+import java.util.ArrayDeque;
+import java.util.concurrent.Flow;
+
+import static org.elasticsearch.xpack.inference.common.DelegatingProcessorTests.onError;
+import static org.elasticsearch.xpack.inference.common.DelegatingProcessorTests.onNext;
+import static org.elasticsearch.xpack.inference.external.response.streaming.StreamingInferenceTestUtils.containsResults;
+import static org.elasticsearch.xpack.inference.external.response.streaming.StreamingInferenceTestUtils.events;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.sameInstance;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class GoogleAiStudioStreamingProcessorTests extends ESTestCase {
+
+    public void testParseSuccess() {
+        var item = events("""
```

Comment:
Would it be possible to just mock this object? I know we usually write out the response objects in our tests so it's fine to leave this as is. I'm wondering though if we're able to reduce our test sizes by mocking some of these responses instead.

Reply:
We cannot, because it's a record/final class, and I don't believe we're running with `mockito-inline` (which allows mocking final classes)
```
Cannot mock/spy class org.elasticsearch.xpack.inference.external.response.streaming.ServerSentEvent
Mockito cannot mock/spy because :
 - final class
```

But I guess that makes sense, because in a way a mock is a record